### PR TITLE
Adds display name subclass for rails values emails

### PR DESCRIPTION
**Pull request by natewallis**
Need a better way to describe emails that contain a display name.  Previous comparisons failed because the original `RailsValues::EmailAddress` failed because it doesn't take into account the display name of an email address.  This PR adds a new subclass called `DisplayNameEmailAddress` to handle this comparison.  I have also renamed any previous descriptions of `FormattedEmailAddress` to `DisplayNameEmailAddress` as I don't think `Formatted` really described what the classes were up to.  Wasn't sure if I needed to, but I kept the original cast type in `railtie.rb` in case it broke production code.

This PR moves all previous display name logic out of `EmailAddress` as the class was not initially designed to handle display names in emails.

### CODE REVIEW

File: lib/rails_values/email_address.rb [hunk 1]
```diff
@@ -14,8 +14,7 @@ def initialize(address)
       freeze
     end
 
-    delegate :local, :display_name, :format, to: :mail_address
-
+    delegate :local, :display_name, to: :mail_address
```
Comment (natewallis) [inline lib/rails_values/email_address.rb:17]: `format` on mail address originally called as part of `EmailAddress` to support display names, but I think it really only belongs in subclass now

File: spec/rails_values/display_name_email_address_spec.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    it 'to_s should lower case the email address' do
+      expect(cast('Test <TEST@mail.com>')).to eq('Test <test@mail.com>')
+    end
+
+    it 'they are equal if display names and emails are the same' do
+      first_cast = cast('Test <test@mail.com>')
+      second_cast = cast('Test <test@mail.com>')
+      expect(first_cast).to eq(second_cast)
+      expect(first_cast).to eql(second_cast)
```
Comment (bapu-nexl) [inline spec/rails_values/display_name_email_address_spec.rb:22]: repeated line?

**Review by bapu-nexl — APPROVED** [review]
looks good

**Review by grantspeelman — COMMENTED** [review]
Nice Work 👍🏽

**Pull request merged**